// mongolink/index.js
var ml = {};

ml.updateData = function(selector, nosql, callback) {
	mongodb = require("mongodb");
	mongodb.MongoClient.connect(ml.uri, {
		server: {
			auto_reconnect: true
		}
	}, function(err, db) {
		db.collection(nosql.collection, function(err, db) {
			db.update(selector, {
				$set: nosql.document
			}, {
				upsert: false,
				w: 1
			}, callback);
		});
	});
};

ml.updateDataArray = function(selector, nosql, callback) {
	mongodb = require("mongodb");
	mongodb.MongoClient.connect(ml.uri, {
		server: {
			auto_reconnect: true
		}
	}, function(err, db) {
		db.collection(nosql.collection, function(err, db) {
			db.update(selector, {
				$addToSet: nosql.document
			}, {
				upsert: false,
				w: 1
			}, callback);
		});
	});
};

ml.insertData = function(nosql, callback) {
	mongodb = require("mongodb");
	mongodb.MongoClient.connect(ml.uri, {
		server: {
			auto_reconnect: true
		}
	}, function(err, db) {
		var collection = new mongodb.Collection(db, nosql.collection);

		collection.insert(nosql.document, {
			safe: true
		}, callback);
	});
};

ml.getData = function(nosql, callback) {

	// var args = Array.prototype.slice.call(arguments, 0),
	// 	hasCallback = typeof args[args.length - 1] === 'function',
	// 	hasWeirdCallback = typeof args[0] === 'function',
	// 	callback = hasCallback ? args.pop() : (hasWeirdCallback ? args.shift() : null),
	// 	len = args.length,
	// 	selector = len >= 1 ? args[0] : {}, fields = len >= 2 ? args[1] : {};
	// 	

	var	selector = nosql.selector || {},
		fields = nosql.fields || {};

	mongodb = require("mongodb");
	mongodb.MongoClient.connect(ml.uri, {
		server: {
			auto_reconnect: true
		}
	}, function(err, db) {
		if (err) return callback(err);

		db.collection(nosql.collection, function(err, collection) {
			if (err) return callback(err);
			console.log("inside getData");
			console.log(selector);
			collection.find(selector, fields).toArray(callback);
		});
	});
};
module.exports = ml;
